Game/Source/GDKTestGyms/Private/NFRConstants.cpp

```diff
@@ -7,26 +7,46 @@
 
 DEFINE_LOG_CATEGORY(LogNFRConstants);
 
-FMetricDelay::FMetricDelay(float Seconds)
+FMetricTimer::FMetricTimer(int32 Seconds)
+	: bLocked(false)
 {
-	SetDelay(Seconds);
+	SetTimer(Seconds);
 }
 
-void FMetricDelay::SetDelay(float Seconds)
+bool FMetricTimer::SetTimer(int32 Seconds)
 {
-	TimeToStart = FDateTime::Now().GetTicks() + FTimespan::FromSeconds(Seconds).GetTicks();
+	if (!bLocked)
+	{
+		TimeToStart = FDateTime::Now().GetTicks() + FTimespan::FromSeconds(Seconds).GetTicks();
+	}
+	return !bLocked;
+}
+
+void FMetricTimer::SetLock(bool bState)
+{
+	bLocked = bState;
 }
 
-bool FMetricDelay::IsReady() const
+bool FMetricTimer::HasTimerGoneOff() const
```

Comment:
Could we set a max time for these? That way we wouldn't need the additional life time timer, and it could be contained in this obj.

Reply:
So HasTimerGoneOff only returns true when we are after an initial delay but before another time. Hmm potentially. That would work. These would no longer be timers. I feel as though this would be harder to understand. I think everyone understands a timer and knows what to expect. If you feel the "code benefit" would outweigh the "understandability" benefit then I will implement this suggestion.

Reply:
It seems like it could be a common pattern - ie. report the metric within this window.
However if this is the only case currently, then I'm fine with it.